### PR TITLE
feat: add resource classes and images for Windows & GPU executors

### PR DESCRIPTION
**Pull request by loderunner**
[Jira](https://circleci.atlassian.net/browse/LIFE-112)

# Description

This PR goes over many problems with autocompletion and validation of machine executors.

# Implementation details

* Remove obsolete `windows:` executors
* Add images and resource classes for many machine executors
* Update Xcode versions and macOS resource classes
* Do some refactoring along the way

# How to validate

There are many cases to validate when writing an executor. Try some valid and invalid cases and see if the error messages make sense.

----

For reference, this is the validation process I came up with for machine executors

image

### CODE REVIEW

**Review by jackieg017 — APPROVED** [review]
LGTM! I'm by no means an expert in modifying executors though 😅

File: go.mod [hunk 1]
```diff
@@ -1,6 +1,6 @@
 module github.com/CircleCI-Public/circleci-yaml-language-server
 
-go 1.21
+go 1.22
```
Comment (abdelDriowya) [inline go.mod:3]: doesn't work I replaced this with `go 1.22.0 `to make it work

**Review by abdelDriowya — CHANGES_REQUESTED** [review]
doesn't seem to work
Screenshot 2024-05-23 at 12 40 09

File: go.mod (hunk 1 above)
Comment (loderunner) [reply go.mod:3]: What do you mean by "doesn't work"? This should only affect build, and build seems to work [in the CI](https://app.circleci.com/pipelines/github/CircleCI-Public/circleci-yaml-language-server/1290/workflows/e9044bf6-3f8a-4803-8a3c-ae8d4cc13517/jobs/7873).

Can you describe what doesn't work? Did you install go 1.22 on your machine?

**Comment by loderunner** [comment]
@abdelDriowya 
> doesn't seem to work Screenshot 2024-05-23 at 12 40 09

I don't think `windows-server-2019:2022.08.1` is a valid windows image. Have you tried running a pipeline with this config file? Can you try with one of the images from the docs, e.g. `windows-server-2019-vs2019:2023.08.1`?

**Comment by abdelDriowya** [comment]
Oh I didn't know about this. 
indeed `windows-server-2019:2022.08.1` is working

File: go.mod (hunk 1 above)
Comment (abdelDriowya) [reply go.mod:3]: @loderunner I think this change is manadatory for LS to build

**Review by abdelDriowya — APPROVED** [review]
Local issue with go version no need to change anything

**Pull request merged**